### PR TITLE
[17.0][MIG] partner_firstname: Migration to 17.0

### PR DESCRIPTION
**Pull request by moitabenfdz**

### CODE REVIEW

**Comment by moitabenfdz** [comment]
Checks not passed because of OCA server issues. Will try to rebase and run test again latter.

**Review by dreispt — APPROVED** [review]

File: partner_firstname/views/base_config_view.xml
Comment (AntoniRomera) [inline partner_firstname/views/base_config_view.xml]:
```suggestion
		<block title="Partner Names Order" name="partner_names_order_setting_container">
			<setting>
				<field name="partner_names_order"  nolabel="1"/>
				<field name="partner_names_order_changed" invisible="1" />
				 <button
                                name="action_recalculate_partners_name"
                                string="Recalculate names"
                                icon="fa-play"
                                type="object"
                                help="Recalculate names for all partners. This process could take so much time if there are more than 10,000 active partners"
                                invisible="not partner_names_order_changed"
                            />
			</setting>
		</block>          
```

**Review by AntoniRomera — APPROVED** [review]
LGTM, With the old xml it works, the change is just to make it more correct for V17, but the module works correctly in a local instance.  With the old xml it works, the change is just to make it more correct for V17, but the module works correctly in a local instance.

File: partner_firstname/static/description/index.html
Comment (yale-dynapps) [inline partner_firstname/static/description/index.html:1]: On Odoo.sh to have the green light on branch, we need to remove this line

**Review by yale-dynapps — CHANGES_REQUESTED** [review]

File: partner_firstname/static/description/index.html
Comment (AntoniRomera) [reply partner_firstname/static/description/index.html:1]: This would have to be said in the mainteners repo as it creates it only when you pass the pre-commit.

**Review by MRomeera — APPROVED** [review]
lgtm

**Comment by AntoniRomera** [comment]
@pedrobaeza  Can you merge it? It works correctly

**Comment by pedrobaeza** [comment]
/ocabot migration partner_firstname

I see a comment unattended in `base_config_view.xml`, isn't it?

**Comment by AntoniRomera** [comment]
> /ocabot migration partner_firstname
> 
> I see a comment unattended in `base_config_view.xml`, isn't it?

Yes, but it does not affect the functionality at all, and as it stands it works fine.

**Comment by pedrobaeza** [comment]
But it's the new syntax to be followed, so it's preferable to be changed. @moitabenfdz can you please change it?

**Comment by pedrobaeza** [comment]
@AntoniRomera can you please extract a text for such conversion of the seetings to be added to the migration guide?

**Comment by AntoniRomera** [comment]
> @AntoniRomera can you please extract a text for such conversion of the seetings to be added to the migration guide?

Issue with the text: [#594](https://github.com/OCA/maintainer-tools/issues/594)

**Comment by moitabenfdz** [comment]
@pedrobaeza Changed it! Thank you @AntoniRomera for the proposed changes.

**Comment by pedrobaeza** [comment]
Please squash the latest commit into the migration one.

**Comment by AntoniRomera** [comment]
@pedrobaeza I think it's more than ready now.

**Comment by rousseldenis** [comment]
@OCA/board @OCA/core-maintainers It seems module is merged but not published on OCA shop:

https://odoo-community.org/shop/partner-first-name-and-last-name-1794?search=partner_firstname#attr=21627

nor on Odoo apps page: https://apps.odoo.com/apps/modules/16.0/partner_firstname/

But correctly on pypi : https://pypi.org/project/odoo-addon-partner-firstname/


Could you look at ?

**Pull request merged**
